File: recursion.js
// recursion.js
function factorial(n) {
    if (n <= 1) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('Enter an integer: ', input => {
    const param = parseInt(input, 10);
    if (isNaN(param)) {
        console.log('Invalid input');
    } else {
        console.log('Factorial: ' + factorial(param));
    }
    readline.close();
});

File: recursion.kt
// recursion.kt
fun factorial(n: Int): Int {
    return if (n <= 1) 1 else n * factorial(n - 1)
}

fun main() {
    print("Enter an integer: ")
    val input = readLine()
    val param = input?.toIntOrNull()
    if (param == null) {
        println("Invalid input")
    } else {
        println("Factorial: ${factorial(param)}")
    }
}
